velox/exec/tests/utils/TpchQueryBuilder.cpp

```diff
@@ -1060,6 +1062,134 @@ TpchPlan TpchQueryBuilder::getQ10Plan() const {
   return context;
 }
 
+TpchPlan TpchQueryBuilder::getQ11Plan() const {
+  // Column definitions
+  std::vector<std::string> partsuppColumns = {
+      "ps_partkey", "ps_suppkey", "ps_availqty", "ps_supplycost"};
+  std::vector<std::string> partsuppColumnsSubQuery = {
+      "ps_availqty", "ps_suppkey", "ps_supplycost"};
+  std::vector<std::string> supplierColumns = {"s_suppkey", "s_nationkey"};
+  std::vector<std::string> nationColumns = {"n_nationkey", "n_name"};
+
+  // Row type definitions
+  auto partsuppRowType = getRowType(kPartsupp, partsuppColumns);
+  auto partsuppRowTypeSubQuery = getRowType(kPartsupp, partsuppColumnsSubQuery);
+  auto supplierRowType = getRowType(kSupplier, supplierColumns);
+  auto nationRowType = getRowType(kNation, nationColumns);
+
+  // File columns
+  const auto& partsuppFileColumns = getFileColumnNames(kPartsupp);
+  const auto& partsuppFileColumnsSubQuery = getFileColumnNames(kPartsupp);
+  const auto& supplierFileColumns = getFileColumnNames(kSupplier);
+  const auto& nationFileColumns = getFileColumnNames(kNation);
+
+  // Filter for nation name
+  const std::string nationNameFilter = "n_name = 'GERMANY'";
+
+  // Plan node ID generator
+  auto planNodeIdGenerator = std::make_shared<core::PlanNodeIdGenerator>();
+
+  // Subquery plan nodes
+  core::PlanNodeId partsuppScanIdSubQuery;
+  core::PlanNodeId supplierScanIdSubQuery;
+  core::PlanNodeId nationScanIdSubQuery;
+
+  // Main query plan nodes
+  core::PlanNodeId partsuppScanId;
+  core::PlanNodeId supplierScanId;
+  core::PlanNodeId nationScanId;
+
+  // Subquery plan to calculate filter cost
+  auto nationSubQuery =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kNation, nationRowType, nationFileColumns, {nationNameFilter})
+          .capturePlanNodeId(nationScanIdSubQuery)
+          .planNode();
+
+  auto supplierJoinNationSubQuery =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(kSupplier, supplierRowType, supplierFileColumns)
+          .capturePlanNodeId(supplierScanIdSubQuery)
+          .hashJoin(
+              {"s_nationkey"},
+              {"n_nationkey"},
+              nationSubQuery,
+              "",
+              {"s_suppkey"})
+          .planNode();
+
+  auto subQueryPlan =
+      PlanBuilder(planNodeIdGenerator)
+          .tableScan(
+              kPartsupp, partsuppRowTypeSubQuery, partsuppFileColumnsSubQuery)
+          .capturePlanNodeId(partsuppScanIdSubQuery)
+          .hashJoin(
+              {"ps_suppkey"},
+              {"s_suppkey"},
+              supplierJoinNationSubQuery,
+              "",
+              {"ps_availqty", "ps_suppkey", "ps_supplycost"})
+          .project({"ps_supplycost * ps_availqty AS product_cost_qty"})
+          .partialAggregation({}, {"sum(product_cost_qty) AS sum_cost_qty"})
+          .localPartition({})
+          .finalAggregation(
+              {}, {"sum(sum_cost_qty) AS filter_cost_qty"}, {{DOUBLE()}})
```

Comment:
Can we just use `finalAggregation()` here? Is it needed to explicitly specify the return type?

Reply:
Yes, we can use finalAggregation() without specifying the return type explicitly. I have updated the code to make it cleaner.